single_cell_RNAseq/README.md

```diff
@@ -94,6 +94,11 @@ In addition to being used as the initial values for your 'qc_cuts.csv', the valu
 
 Of note, it can be useful to adjust these values sensibly before / for all runs by adjusting this file.  This can be especially useful with tissue data where it is often useful to compare across libraries and batches, with consistent cutoff values, as a method of assessing relative quality across the entire project.
 
+### c4 partitions, additional configuration
+We have the following partitions available on the c4 cluster for DSCoLab members: krummellab,common (default partitions, which are set in the environment variable `$SBATCH_PARTITION`) and freecycle. Freecycle allows us to run jobs on any compute that is not in use (24h job max), with the caveat that the job may be cancelled if a partition owner requires these resources. We have set the default for `test` jobs to freecycle, and for standard jobs to `freecycle,krummellab,common`, which means they are first submitted to freecycle, then krummellab, then common. You may want to adjust this and other cluster options, such as the errorStrategy (what happens to the whole pipeline when it errors). See comments in `nextflow.config` describing what we have set as defaults, and what you may want to adjust.
```

Comment:
After "when it errors)." and before "See comments in `nextflow.config`", perhaps add:

If you do not have access to the krummellab partition on c4, you will need to change the `process.queue` parameter in your `nextflow.config` file.

Reply:
Thankfully, it just defaults to whichever ones you have access to. So if I submit to costellolab,freecycle, it just goes to freecycle. Similarly if the resources are too high for freecycle (e.g. cellranger requests > 24h), it goes to common,krummellab

Reply:
Added the note to the README too, thanks!

Reply:
> Just a note: I wonder what nextflow does if a job gets cancelled by the scheduler? We might want to build some sort of catch and retry in if possible... But since we've rarely, if ever, seen freecycle jobs actually get cancelled from primary lab prioritization, probably fine to not build this until needed
I think we should definitely do this! There is a way to set it to retry on specific job codes, e.g.:
`errorStrategy { task.exitStatus == 140 ? 'retry' : 'terminate' }`
Let me see if I can figure out how to figure out what the job code is when jobs are cancelled by freecycle, and then I can add this... Let me know if you have examples.

Reply:
> Thankfully, it just defaults to whichever ones you have access to. So if I submit to costellolab,freecycle, it just goes to freecycle. Similarly if the resources are too high for freecycle (e.g. cellranger requests > 24h), it goes to common,krummellab

Oh nice!  Then... maybe the note I had you add isn't needed.  Or at least as is.  Cuz it works out fine if they don't change it!

Reply:
> default number of jobs as 60 seems pretty high to me? 60 cellranger jobs would take A LOT of resources.

Noted - lowering to 20 for single cell pipeline. I wish there were a way to set a max number of cpus grrr because this is more what we want. I'll ask on the nextflow slack.
We should all discuss @AlaaALatif (this is what is set for bulk), and also bring up at a group meeting what folks want for this or other defaults

Reply:
> Oh nice! Then... maybe the note I had you add isn't needed. Or at least as is. Cuz it works out fine if they don't change it!

I think it's good to note that they might want to change it, happy to keep this :)